pkg/logql/parser.go

```diff
@@ -22,6 +23,8 @@ var parserPool = sync.Pool{
 	},
 }
 
+const maxInputSize = 5120
```

Comment:
why 5K?

```suggestion
const maxInputSize = 5 * (1 << 10) // 5K
```

Reply:
5k is mostly arbitrary, the max a query param should be in a URL is 2k so this was a number bigger than that but still with some extra room

Reply:
To my knowledge nothing enforces that 2KB limit; also I believe one can POST queries which removes that consideration.